server/src/github/github.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common';
+import { CommitConfig, GithubLink } from './github.interface';
+var github = require("octonode");
+
+@Injectable()
+export class GithubService {
+	testDatabase = {
+		repo: {
+			1: "spencer012/Game-8",
+			2: "d-oliveros/nest"
+		}
+	}
+
+	client = github.client();
+
+	/*
+		returns all commits from the repo specified by the id, optionally by author
+	*/
+	async getCommits(config: CommitConfig): Promise<Array<object>> {
+		let commitOptions: {per_page, page, author?} = { per_page: 100, page: 1 };
+		if (config.author) { 
+			commitOptions.author = config.author;
+		}
+		
+		let repo = this.client.repo(this.testDatabase.repo[config.repo]);
+		let responseData = await repo.commitsAsync(commitOptions); //get first page
+		//console.log(responseData);
+		let commits: Array<any> = responseData[0];
+
+		if (responseData[1].link) { //if it has pagination
```

Comment:
For the `responseData[1]`, I think it is possible to do Array restructuring. Something like:

```
//let us say you have 
const parson = ["Gilberto", "Diaz"]
const [firstName, lastName] = parson
```

Reply:
Resolved with [commit](https://github.com/Lunatic-Labs/LU-Overview/pull/24/commits/ad7f4d12eb8305923cd43330e1912a0c4cf70dfb)